modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountExpirationTest.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.discount.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.commerce.discount.constants.CommerceDiscountConstants;
+import com.liferay.commerce.discount.model.CommerceDiscount;
+import com.liferay.commerce.discount.service.CommerceDiscountLocalService;
+import com.liferay.petra.function.UnsafeRunnable;
+import com.liferay.portal.kernel.scheduler.SchedulerJobConfiguration;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.math.BigDecimal;
+
+import java.util.Calendar;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Balazs Breier
+ */
+@RunWith(Arquillian.class)
+public class CommerceDiscountExpirationTest {
+
+	@ClassRule
+	@Rule
+	public static AggregateTestRule aggregateTestRule = new AggregateTestRule(
+		new LiferayIntegrationTestRule(),
+		PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testViewExpiredDiscountEntry() throws Exception {
+		CommerceDiscount commerceDiscount1 =
+			_commerceDiscountLocalService.addCommerceDiscount(
+				TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+				CommerceDiscountConstants.TARGET_PRODUCT_GROUPS, false, null,
+				true, BigDecimal.ONE, CommerceDiscountConstants.LEVEL_L1,
+				BigDecimal.ONE, BigDecimal.ZERO, BigDecimal.ZERO,
+				BigDecimal.ZERO, CommerceDiscountConstants.TYPE_PERCENTAGE, 0,
+				false, true, 1, 1, 2020, 1, 1, 1, 1, 2020, 1, 1, false,
+				ServiceContextTestUtil.getServiceContext());
+
+		CommerceDiscount commerceDiscount2 =
+			_commerceDiscountLocalService.addCommerceDiscount(
+				TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+				CommerceDiscountConstants.TARGET_PRODUCT_GROUPS, false, null,
+				true, BigDecimal.ONE, CommerceDiscountConstants.LEVEL_L1,
+				BigDecimal.ONE, BigDecimal.ZERO, BigDecimal.ZERO,
+				BigDecimal.ZERO, CommerceDiscountConstants.TYPE_PERCENTAGE, 0,
+				false, true, 1, 1, 2020, 1, 1, 1, 1, 2020, 1, 1, false,
+				ServiceContextTestUtil.getServiceContext());
+
+		commerceDiscount1 = _commerceDiscountLocalService.getCommerceDiscount(
```

Comment:
Why using a get here? Didn't the add method returned the same object?

Reply:
@matyaswollner yeah, it should. I like to 'get' it after to make sure that is retrieved by the proper function and it is easier to read for me. Somewhat redundant but for me it is more precise.